DiscordChatExporter.Core/Exporting/Writers/HtmlMessageWriter.cs

```diff
@@ -28,6 +28,13 @@ public HtmlMessageWriter(Stream stream, ExportContext context, string themeName)
     private bool CanJoinGroup(Message message) =>
         // First message in the group can always join
         _messageGroup.LastOrDefault() is not { } lastMessage ||
+        // Group system messages only with eachother
+        ( 
+            message.Kind.IsSystemMessage() &&
+            lastMessage.Kind.IsSystemMessage()
+        ) ||
+        // Join non system messages
+        !message.Kind.IsSystemMessage() &&
```

Comment:
Can we simplify this somehow by moving the system message related checks to an additional check before it (with a return statement)? It's getting a little confusing